internal/provider/project_environment_resource_test.go

```diff
@@ -0,0 +1,165 @@
+						environment_name = unleash_environment.space.name
+						enabled = true
+						change_requests_enabled = true
+						required_approvals = 20
+					}
+				`,
+				ExpectError: regexp.MustCompile(`(?s)Error: Invalid required_approvals value.*The required_approvals attribute must be between 1 and 10, but got: 20`),
+			},
+			{
+				Config: `
+					resource "unleash_project" "galaxy-wide-energy" {
+						id = "dysonsphere"
+						name = "dysonsphere"
+					}
+
+					resource "unleash_environment" "space" {
+						name = "outerspace"
+						type = "vacuum"
+					}
+
+					resource "unleash_project_change_request" "approvals" {
+						project_id = unleash_project.galaxy-wide-energy.id
+						environment_name = unleash_environment.space.name
+						enabled = true
+						change_requests_enabled = true
+						required_approvals = 0
+					}
+				`,
+				ExpectError: regexp.MustCompile(`(?s)Error: Invalid required_approvals value.*The required_approvals attribute must be between 1 and 10, but got: 0`),
+			},
+			{
+				Config: `
+					resource "unleash_project" "galaxy-wide-energy" {
+						id = "dysonsphere"
+						name = "dysonsphere"
+					}
+
+					resource "unleash_environment" "space" {
+						name = "outerspace"
+						type = "vacuum"
+					}
+
+					resource "unleash_project_change_request" "approvals" {
+						project_id = unleash_project.galaxy-wide-energy.id
+						environment_name = unleash_environment.space.name
+						enabled = true
+						change_requests_enabled = false
+					}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("unleash_project_change_request.approvals", "enabled", "true"),
+					resource.TestCheckResourceAttr("unleash_project_change_request.approvals", "change_requests_enabled", "false"),
+					resource.TestCheckNoResourceAttr("unleash_project_change_request.approvals", "required_approvals"),
+				),
+			},
+			{
+				Config: `
+					resource "unleash_project" "galaxy-wide-energy" {
+						id = "dysonsphere"
+						name = "dysonsphere"
+					}
+
+					resource "unleash_environment" "space" {
+						name = "outerspace"
+						type = "vacuum"
+					}
+
+					resource "unleash_project_change_request" "approvals" {
+						project_id = unleash_project.galaxy-wide-energy.id
+						environment_name = unleash_environment.space.name
+						enabled = false
+					}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("unleash_project_change_request.approvals", "enabled", "false"),
+					resource.TestCheckResourceAttr("unleash_project_change_request.approvals", "change_requests_enabled", "false"),
+				),
+			},
+			{
+				Config: `
+					resource "unleash_project" "galaxy-wide-energy" {
+						id = "dysonsphere"
+						name = "dysonsphere"
+					}
+
+					resource "unleash_environment" "testing" {
+						name = "lab-environment"
+						type = "testing-environment"
+					}
+
+					resource "unleash_project_change_request" "approvals" {
```

Comment:
Is the resource named unleash_project_change_request or should it be named unleash_project_settings? I'm wondering because besides the `change_request_enabled` configuration there's also `enabled=true` which I assume is having that environment in the project.

Reply:
Oh no. This is a hangover from an experiment that went wrong where I was going to split this into two resources. This should be named unleash_project_environment resource - it's effectively a join resource that bridges project environment settings. Good spot, thank you!